src/tpm2.c

```diff
@@ -69,11 +69,14 @@ static TPM_RC TPM2_AcquireLock(TPM2_CTX* ctx)
             return TPM_RC_FAILURE;
         }
         ctx->hwLockInit = 1;
+        ctx->lockCount = 0;
     }
 
-    ret = wc_LockMutex(&ctx->hwLock);
-    if (ret != 0)
-        return TPM_RC_FAILURE;
+    if (++ctx->lockCount == 1) {
```

Comment:
Operations inside conditionals should be avoided due to potential compiler optimization issues. Applies to other lines below also.